### PR TITLE
Kucoin: Add order execution limits

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Adds order execution limits

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/kucoin/kucoin_wrapper.go [hunk 1]
```diff
@@ -1564,3 +1564,40 @@ func (ku *Kucoin) GetFuturesContractDetails(ctx context.Context, item asset.Item
 	}
 	return resp, nil
 }
+
+// UpdateOrderExecutionLimits updates order execution limits
+func (ku *Kucoin) UpdateOrderExecutionLimits(ctx context.Context, a asset.Item) error {
```
Comment (gloriousCode) [inline exchanges/kucoin/kucoin_wrapper.go:1569]: Please add this to `Run()` like all other `UpdateOrderExecutionLimits()` implementations

**Review by gloriousCode — CHANGES_REQUESTED** [review]
🆙 only

File: exchanges/kucoin/kucoin_wrapper.go [hunk 2]
```diff
@@ -1564,3 +1576,40 @@ func (ku *Kucoin) GetFuturesContractDetails(ctx context.Context, item asset.Item
...
+	if !ku.SupportsAsset(a) {
+		return fmt.Errorf("%w %v", asset.ErrNotSupported, a)
+	}
+
+	if a != asset.Spot {
```
Comment (gloriousCode) [inline exchanges/kucoin/kucoin_wrapper.go:1586]: I'm going to request that you add all supported assets, all API endpoints have been implemented

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for adding it to `Run()` 🎉

File: exchanges/kucoin/kucoin_test.go [hunk 3]
```diff
@@ -81,14 +79,16 @@ func TestMain(m *testing.M) {
...
 	if err != nil {
 		t.Error("GetSymbols() error", err)
 	}
-	_, err = ku.GetSymbols(context.Background(), currency.BTC.String())
+	assert.NotEmpty(t, symbols, "should return all available spot/margin symbols")
+	symbols, err = ku.GetSymbols(context.Background(), "ETF")
 	if err != nil {
 		t.Error("GetSymbols() error", err)
 	}
+	assert.NotEmpty(t, symbols, "should return all available futures symbols")
```
Comment (gloriousCode) [inline exchanges/kucoin/kucoin_test.go:91]: Did you have an issue where symbols weren't returned, but there was no error?

File: exchanges/kucoin/kucoin_test.go [hunk 4]
```diff
@@ -2436,3 +2436,33 @@ func TestGetFuturesContractDetails(t *testing.T) {
...
+
+func TestUpdateOrderExecutionLimits(t *testing.T) {
+	t.Parallel()
+
+	err := ku.UpdateOrderExecutionLimits(context.Background(), asset.Binary)
+	if !errors.Is(err, asset.ErrNotSupported) {
+		t.Fatalf("Received %v, expected %v", err, asset.ErrNotSupported)
+	}
+
+	assets := []asset.Item{asset.Margin}
```
Comment (gloriousCode) [inline exchanges/kucoin/kucoin_test.go:2448]:
```suggestion
	assets := []asset.Item{asset.Spot, asset.Futures, asset.Margin}
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice work shizbiz, just one real tiny thing from me

File: exchanges/kucoin/kucoin_test.go (hunk 3 above)
Comment (shazbert) [reply exchanges/kucoin/kucoin_test.go]: Added commentary [here](https://github.com/thrasher-corp/gocryptotrader/pull/1369/commits/3c2ea05c069523acc3548970e26327e880f36c7c)

No it was just to test the usage, I thought originally it was the case.

**Review by gloriousCode — APPROVED** [review]
shazbertACK!

**Review by thrasher- — APPROVED** [review]
Nice stuff, indeed this should be forecefully implemented per new exchange implementation given its necessity for trading!

**Pull request merged**